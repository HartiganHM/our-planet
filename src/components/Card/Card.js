import React from 'react';
import { Link } from 'react-router-dom';
import { object } from 'prop-types';

import './Card.scss';

const Card = (props) => {
  const { name, image } = props.animalData;

  return (
    <Link className="card" to={`/our-planet/animals/${name}`}>
      <img
        className="card__animal-image"
        src={image}
        alt={`${name}-endangered-species`}
      />
      <span className="card__animal-name">{name}</span>
    </Link>
  );
};

export default Card;

Card.propTypes = {
  animalData: object,
};
