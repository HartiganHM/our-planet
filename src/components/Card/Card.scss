//Layout, Box, Visual, Typography, Misc
@import '../../styles/mixins.scss';
@import '../../styles/variables.scss';

.card {
  display: flex;
  flex-direction: column;
  @include width(29.8);
  @include margin(1);
  border: $simple-border;
  background: $accent-fade;
  color: $accent-color-dark;
  text-decoration: none;
  overflow: hidden;
  transition: 0.2s color, background ease-in-out;
  &:hover {
    background: $accent-dark-fade;
    color: $snow;
  }
  &__animal-image {
    @include width(30);
    z-index: 2;
  }
  &__animal-name {
    @include padding(1);
    background: {
      image: url('../../images/branding/Our-Planet-Light-Logo.png');
      position: calc(100% + 5px) calc(100% - 5px);
      size: contain;
      repeat: no-repeat;
    }
    font-family: $accent-font;
    font-size: 1.2rem;
  }
}
