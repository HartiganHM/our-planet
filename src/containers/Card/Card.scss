.Card {
  width: 30rem;
  height: 30rem;
  .animal-image {
    width: 30rem;
  }
}