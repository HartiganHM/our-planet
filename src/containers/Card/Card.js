import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import './Card.css';

const Card = props => {
  const { name, image } = props.animalData;
  const animalLink = '/animals/' + name;

  return (
    <Link to={animalLink} className='Card'>
      <img
        className='animal-image'
        src={require(`../../images/animals/${image}.jpg`)}
        alt={`${name}-endangered-species`}
      />
      <span className='animal-name'>{name}</span>
    </Link>
  );
};

export default Card;

Card.propTypes = {
  animalData: PropTypes.object
};
