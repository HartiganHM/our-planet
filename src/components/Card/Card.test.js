import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card tests', () => {
  it('Should exist', () => {
    const renderedCard = shallow(<Card />);
    expect(renderedCard).toBeDefined();
  });
});
