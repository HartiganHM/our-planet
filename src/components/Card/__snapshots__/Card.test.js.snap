// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    animalData={
        Object {
            "continent_id": 2,
            "facts": "The Arctic fox is primarily a carnivore that lives inland, away from the coasts. They are dependent on the presence of smaller animals (most often lemmings) to survive.  Arctic foxes also hunt for sea birds, fish, and other marine life. Smaller rodent populations waver between times of abundance and scarcity, which leaves the Arctic fox vulnerable when these creatures are low in numbers.",
            "habitat": "Antarctica / Arctic",
            "height": "11 inches",
            "human_benefit": "The protection of the Arctic fox ensures the safety of a variety of other wildlife in the Arctic region.",
            "image": "arctic fox",
            "length": "18 to 27 inches",
            "name": "Arctic Fox",
            "population": "Several Hundred Thousand",
            "scientific_name": "Vulpes lagopus",
            "status": "Least Concern",
            "weight": "3 to 20 pounds",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <img
          alt="Arctic Fox-endangered-species"
          className="animal-image"
          src="arctic fox.jpg"
/>,
        <span
          className="animal-name"
>
          Arctic Fox
</span>,
      ],
      "className": "Card",
      "replace": false,
      "to": "/our-planet/animals/Arctic Fox",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Arctic Fox-endangered-species",
          "className": "animal-image",
          "src": "arctic fox.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Arctic Fox",
          "className": "animal-name",
        },
        "ref": null,
        "rendered": "Arctic Fox",
        "type": "span",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <img
            alt="Arctic Fox-endangered-species"
            className="animal-image"
            src="arctic fox.jpg"
/>,
          <span
            className="animal-name"
>
            Arctic Fox
</span>,
        ],
        "className": "Card",
        "replace": false,
        "to": "/our-planet/animals/Arctic Fox",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Arctic Fox-endangered-species",
            "className": "animal-image",
            "src": "arctic fox.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Arctic Fox",
            "className": "animal-name",
          },
          "ref": null,
          "rendered": "Arctic Fox",
          "type": "span",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
